fast_barcode_scanner/android/src/main/kotlin/com/jhoogstraat/fast_barcode_scanner/cache/ImageHelper.kt

```diff
@@ -0,0 +1,102 @@
+import android.content.Context
+import android.graphics.ImageFormat
+import android.graphics.Rect
+import android.graphics.YuvImage
+import android.media.Image
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+
+class ImageHelper {
+    private var savedCodes: MutableMap<String, String> = mutableMapOf()
+
+    private object Holder {
+        val INSTANCE = ImageHelper()
+    }
+
+    companion object {
+        @JvmStatic
+        fun getInstance(): ImageHelper {
+            return Holder.INSTANCE
+        }
+    }
+
+    // Store image to the path with barcode as filename
+    private fun storeImage(imageBytes: ByteArray?, key: String, context: Context) {
+        val externalFilesDirectory = context.cacheDir;
+        val imageFile: File
+
+        try {
+            externalFilesDirectory.mkdirs()
+            val barcodeDirectory = File(externalFilesDirectory, "barcode_images")
+            if (!barcodeDirectory.exists()) {
+                barcodeDirectory.mkdirs()
+            }
+
+            imageFile = File.createTempFile(key, ".jpeg", barcodeDirectory)
+            imageFile.deleteOnExit()
```

Comment:
we probably don't want this deleteOnExit behavior. The file may not be uploaded in this session. the background file uploader may upload this file much later.